pandas/core/nanops.py

```diff
@@ -1194,3 +1194,38 @@ def f(x, y):
 nanle = make_nancomp(operator.le)
 naneq = make_nancomp(operator.eq)
 nanne = make_nancomp(operator.ne)
+
+
+def _nanpercentile1D(values, mask, q, na_value, interpolation):
+    # mask is Union[ExtensionArray, ndarray]
```

Comment:
can you add doc-strings

Reply:
just pushed with docstrings.  We're going to simplify the tar out of these methods if/when #24600 gets fixed.